Overview.bs

```diff
@@ -623,13 +626,479 @@ The algorithm:
+      [=Mapping Entry/format=] bit 0 is set.
+    </td>
+  </tr>
+
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Mapping Entry">designSpaceCount</dfn></td>
+    <td>
+      Number of elements in the design space list. Only present if [=Mapping Entry/format=] bit 0 is set.
+    </td>
+  </tr>
+  <tr>
+    <td>[=Design Space Segment=]</td>
+    <td><dfn for="Mapping Entry">designSpaceSegments</dfn>[designSpaceCount]</td>
+    <td>
+      List of design space segments in the entry's [=font subset definition=]. Only present if [=Mapping Entry/format=] bit 0 is set.
+    </td>
+  </tr>
+
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Mapping Entry">copyCount</dfn></td>
+    <td>
+      Number of entries in the copyIndices list. Only present if [=Mapping Entry/format=] bit 1 is set.
+    </td>
+  </tr>
+  <tr>
+    <td>uint16</td>
+    <td><dfn for="Mapping Entry">copyIndices</dfn>[copyCount]</td>
+    <td>
+      List of indices from the [=Mapping Entries/entries=] array whose [=font subset definition=] should be copied into this entry. All
+      values must be less than the index of this [=Mapping Entry=] in [=Mapping Entries/entries=]. Only present if
+      [=Mapping Entry/format=] bit 1 is set.
+    </td>
+  </tr>
+
+  <tr>
+    <td>int16</td>
+    <td><dfn for="Mapping Entry">entryIdDelta</dfn></td>
+    <td>
+      Signed delta which is added to the entry id of the previous [=Mapping Entry=] to produce the id for this entry.
+      Only present if [=Mapping Entry/format=] bit 2 is set.
+    </td>
+  </tr>
+
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Mapping Entry">patchEncoding</dfn></td>
+    <td>
+      Specifies the format of the patch linked to by this entry. Uses the ID numbers from the [[#font-patch-formats-summary]] table.
+      Overrides [=Format 2 Patch Map/defaultPatchEncoding=].
+      Only present if [=Mapping Entry/format=] bit 3 is set.
+    </td>
+  </tr>
+
+
+  <tr>
+    <td>uint16/uint24</td>
+    <td><dfn for="Mapping Entry">bias</dfn></td>
+    <td>
+      Bias value which is added to all codepoint values in the codepoints set.
+      If format bit 4 is 0 and bit 5 is 1, then this is present and a uint16.
+      If format bit 4 is 1 and bit 5 is 1, then this is present and a uint24.
+      Otherwise it is not present.
+    </td>
+  </tr>
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Mapping Entry">codepoints</dfn>[variable]</td>
+    <td>
+      Set of codepoints for this mapping. Encoded as a [[#sparse-bit-set-decoding]]. 
+      Only present if [=Mapping Entry/format=] bit 4 and/or 5 is set. The length is
+      determined by following the decoding procedures in [[#sparse-bit-set-decoding]].
+    </td>
+  </tr>
+</table>
+
+<dfn>Design Space Segment</dfn> encoding:
+
+<table>
+  <tr>
+    <th>Type</th><th>Name</th><th>Description</th>
+  </tr>
+  <tr>
+    <td>Tag</td>
+    <td><dfn for="Design Space Segment">tag</dfn></td>
+    <td>
+      Axis tag value.
+    </td>
+  </tr>
+
+  <tr>
+    <td>Fixed</td>
```

Comment:
What's the point for Fixed for these? Why not F2DOT14? Is there some purpose for the increased granularity? Are you thinking these are in user units, so everything has to go through fvar and avar? That might be untenable when avar2 comes into the picture.

Reply:
Yes I've so far been treating this as being in the user units so I chose Fixed to match what fvar uses. The advantage to user units is the client/browser will be operating in user units and then won't need to know about or process internal font variation details to match mapping entries and apply patches. I'll see if I can add some text somewhere in the spec to make it explicit that throughout we're dealing in the external axis coordinates not the internal normalized ones.

I'm not super familiar with avar2, could you expand on what problems it might cause?

Reply:
Hmm. I suppose one can always include a superset of the axis information if one needs it. 

I recommend chatting a bit with Behdad a bit about this aspect of the project. He'll be able to give opinions on whether avar2 could cause problems.

Reply:
The short answer, from what I understand and remember, is that the mapping through avar 2 can be quite complex, leading to gaps. But presumably the subsetter would just *include* those gaps, so maybe it isn't a big deal.

Reply:
Sounds good, I'll follow up with Behdad. My take is that this is will likely be a problem for subsetter implementations to figure out (supersetting to cover gaps seems reasonable to me) and if we stick to user axis space here, it hopefully won't be something we specifically need to address in this spec.

Reply:
To follow up on this Behdad has started to do some work on planning out how subsetting might work for avar2. Don't have a general solution yet, but sounds like we should be able to get to something workable for most cases. After some more though we'll definitely need to stick to user space units on the IFT side since if the font is initially static (with option to augment design space) then the client won't have any information on the mapping from user space to normalized space.